include/private/gc_priv.h

```diff
@@ -2270,6 +2270,23 @@ ptr_t GC_save_regs_in_stack(void);
 #  define LOAD_PTR_OR_CONTINUE(v, p) (void)(v = *(ptr_t *)(p))
 #endif /* !CHERI_PURECAP */
 
+#ifdef CHERI_PURECAP
+# define VALID_CAPABILITY(cap, base_addr)                                   \
+  (cheri_tag_get(cap) && ADDR(cap) >= base_addr                             \
+      && ADDR(cap) < base_addr + cheri_length_get(cap)                      \
+      && (cheri_perms_get(cap) & (CHERI_PERM_LOAD | CHERI_PERM_LOAD_CAP))   \
+              != 0)
+# define INBOUND_CAPABILITY(h)                                         \
```

Comment:
I think this should be renamed from `INBOUND_CAPABILITY` to something which expresses the purpose of the macro better.  As far as I can see, there are two things going on 
1. check if  calculated capability address is valid and in-bounds. 
2. If yes return the newly calculated bounds value. If NOT, then : 
     a. attempt to obtain a valid capability with the *originally* out-of-bounds capbility from the correct header. If this succeeds, h will be set correctly. 
     b. If 2.a FAILS, then return the invalid capability (We are being optimistic here, but so far, every other location that will utilize the return from this function seems to be making a call to ascertain the *HDR* based on the return value 
     
I think it should be renamed to something like `RESOLVE_CAP()`